### PR TITLE
feat: create `@sequelize/cli` as a replacement for `sequelize-cli`

### LINKED ISSUE #13875: Add scope to CLI project
Release new version of CLI under @sequelize/cli and (optionally) remove sequelize-cli as exported binary

### PR DESCRIPTION
**Pull request by ephys**
## Description of Changes

Closes #13875

This PR adds the basics of the new `@sequelize/cli`. It currently supports a single command: `yarn sequelize migration:generate`

This command can generate the migration in 4 formats: `sql`, `typescript` (with `import`), `cjs` or `esm`.

## Feature showcase

The cli automatically prompts when an input is not provided:

image

Unless `--no-interactive` is provided, in which case it will throw an error for missing flags:

image

This CLI also supports `--json`, which restricts the stdout to JSON data

image

And `--help`

image

## TODO

- [x] Wait for #17190 to be merged
- [x] Add unit tests for `migration:generate`
- [x] Create README.md

### CODE REVIEW

**Review by WikiRik — COMMENTED** [review]
Nice work!

File: packages/cli/src/commands/migration/generate.test.ts [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+const __dirname = fileUrlToDirname(import.meta.url);
+const packageRoot = path.join(__dirname, '..', '..', '..');
+
+function oclifTest() {
+  return test.loadConfig({
+    root: packageRoot,
+  });
+}
+
+describe('migration:generate', () => {
```
[the remark this answers is not in the document]
Comment (ephys) [reply packages/cli/src/commands/migration/generate.test.ts:15]: Added

File: packages/cli/README.md [hunk 2]
```diff
@@ -0,0 +1,29 @@
+<p align="center"><img src="https://raw.githubusercontent.com/sequelize/sequelize/ec80c6252ac500df9342816b7f49957f3974e882/logo.svg" width="100" alt="Sequelize logo" /></p>
+<h1 align="center" style="margin-top: 0;"><a href="https://sequelize.org">Sequelize</a></h1>
+
+This library contains model attribute validators built on top of [validator.js](https://www.npmjs.com/package/validator).
+Read more about model validation in the [Sequelize documentation](https://sequelize.org/docs/v7/core-concepts/validations-and-constraints/).
```
[the remark this answers is not in the document]
Comment (ephys) [reply packages/cli/README.md]: Removed

**Review by WikiRik — APPROVED** [review]

File: packages/cli/package.json [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+    "topics": {
+      "migration": {
+        "description": "Commands for managing database migrations"
+      }
+    }
+  },
+  "repository": "https://github.com/sequelize/sequelize",
+  "scripts": {
+    "build": "rm -rf lib && tsc --project tsconfig.build.json",
+    "postpack": "rm -f oclif.manifest.json",
```
Comment (WikiRik) [inline packages/cli/package.json:59]: I was testing this on my Windows machine as well, `rm -rf` does not work there. Not sure why this is not something picked up by the CI. Maybe there's a difference between Windows 11 and Windows Server?

Comment (ephys) [reply packages/cli/package.json:59]: It's not picked up by the CI because we don't release from windows and we only pack as part of the release process

**Review by WikiRik — APPROVED** [review]
Will do another check on my Windows machine later, but we can merge before that

**Pull request merged**